### PR TITLE
Fix SpatialAverager to account for variable input grid cells areas

### LINKED ISSUE #126: SpatialAverager yields weights that are different from Regridder
With a global rectilinear source grid (lon, lat) and a destination grid with a single tile, I can use Regridder to get weights. 

If instead I use SpatialAverager, using a square polygon with the same boundaries as the single tile above, I get different weights. In particular, it seems like the weights only include the fraction of the source grid covering the destination grid, without accounting for the source grid area. 

I would have expected both sets of weights to be identical.

### PR DESCRIPTION
**Pull request by huard**
Fix #126

The original implementation used to compute the grid weights over
ds_in : polygon mesh
ds_out : source grid
and then rename out_dim and in_dim to in_dim and out_dim. This simplified the code's logic but had the effect of neglecting variations in source grid cell areas. 

The fix sets source and destination to their expected order, but needs to grab the destination field area from the regrid object to perform a necessary renormalization between exteriors and interiors. This required a bit of refactoring which hopefully will clarify what's going on. 

A test was added to make sure weights computed using an output mesh and an output grid are nearly identical.

### CODE REVIEW

**Comment by huard** [comment]
The RtD build is broken, and I had to pin esmfpy and shapely to pass all tests. 
If we're not able to unpin those in this PR, I'll open an issue for this.

File: xesmf/frontend.py [hunk 1]
```diff
@@ -925,6 +932,7 @@ def __init__(
         ----------
         This approach is inspired by `OCGIS <https://github.com/NCPP/ocgis>`_.
         """
+        # Note, I suggest we refactor polys -> geoms
```
Comment (aulemahal) [inline xesmf/frontend.py:935]: I have no objection.

**Review by aulemahal — APPROVED** [review]
This is much clearer that before!

File: xesmf/frontend.py (hunk 1 above)
Comment (huard) [reply xesmf/frontend.py:935]: Ok, let's consider that in a future refactor.

**Pull request merged**